ACE/ace/config-android.h

```diff
@@ -118,6 +118,15 @@
 #  define ACE_LACKS_WCSTOLL
 #  define ACE_LACKS_WCSTOULL
 #  define ACE_LACKS_CONDATTR_SETCLOCK
+#  if (defined ACE_HAS_CONDATTR_SETCLOCK)
```

Comment:
Why undef them here and in some other config files, how can it be that these are already defined?

Reply:
Because in all these cases these are after `config-posix.h`, which could be defining them.

Reply:
Why would config-posix.h define these? I would assume the underlying defines are not set, or are they set incorrectly by the os headers?

Reply:
https://github.com/DOCGroup/ACE_TAO/blob/dcbb7778f1c86c562efbf4e87f404a11592f3a48/ACE/ace/config-linux.h#L147-L155

Reply:
That comment/code is specifically for some linux kernel revisions, I am not sure if this is valid for android/hpux, I wouldn't expect that those set the incorrect posix macros and whether we need to do that for those also.

Reply:
Again the common block could be moved to config-macros.h with an "#if defined (ACE_LACKS_CONDATTR_SETCLOCK)"/"#endif" bracket.

Reply:
I originally had the common code to handle this in `config-all.h`, but `config-macros.h` looks much more appropriate for this.